src/client/clientErrorHandler.ts

```diff
@@ -23,12 +33,24 @@ export class ClientErrorHandler implements ErrorHandler {
 
   closed(): CloseAction {
     this.restarts.push(Date.now());
+    const heapProfileGlob = new glob.GlobSync(`${this.heapDumpFolder}/java_*.hprof`);
+    if (heapProfileGlob.found.length) {
+      // Only clean heap dumps that are generated in the default location.
+      // The default location is the extension global storage
+      // This means that if users change the folder where the heap dumps are placed,
+      // then they will be able to read the heap dumps,
+      // since they aren't immediately deleted.
+      cleanUpHeapDumps(this.context);
+      Telemetry.sendTelemetry(Telemetry.JAVA_OOM_EVT);
```

Comment:
I'd send the Xmx value that triggered the OOME

Reply:
We should be able to collect Xmx from the server startup event, although it might be a good idea to collect it here to catch if they allocate so little RAM that the server can't start.

Reply:
Actually, I think the value from the startup telemetry is better. The user could have changed the value of Xmx in their settings after the server started. The Xmx from server startup telemetry will give the actual Xmx used by the JVM.

Reply:
It's infinitely easier to understand the failure on our backend, if the Xmx value is attached to the error. It'll likely be correct in 99+% of cases

Reply:
Okay. This should be collected in this PR now